### ISSUE TITLE
Define minimum version requirement for Scipy

### ISSUE THREAD

**Issue by bsipocz** [astropy/astropy#10029]
While testing the minimum version requirements for astroML I run into the issue that for astropy, we don't have a minimum Scipy required, e.g this is with astropy 3.0 and scipy 0.16. Even there it seems that at least scipy 0.18 is required.

To avoid this and similar issues in the future, I suggest that we declare minimum dependency for all dependencies, and have a job in CI that tests those versions.

```
Package versions: 
Numpy: 1.13.0
Scipy: 0.16.0
Matplotlib: 3.0.0
Cython: not available
Astropy: 3.0
scikit-learn: 0.18
astroML_addons: not available
```
```
____________________________ ERROR collecting astroML/time_series/tests/test_periodogram.py ____________________________
astroML/time_series/periodogram.py:6: in <module>
    from astropy.timeseries import LombScargle
E   ImportError: No module named 'astropy.timeseries'

During handling of the above exception, another exception occurred:
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
astroML/time_series/__init__.py:1: in <module>
    from .ACF import ACF_scargle, ACF_EK
astroML/time_series/ACF.py:6: in <module>
    from .periodogram import lomb_scargle
astroML/time_series/periodogram.py:8: in <module>
    from astropy.stats import LombScargle
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/stats/__init__.py:18: in <module>
    from .circstats import *
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/stats/circstats.py:14: in <module>
    from astropy.units import Quantity
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/units/__init__.py:14: in <module>
    from .quantity import *
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/units/quantity.py:29: in <module>
    from .quantity_helper import (converters_and_unit, can_have_arbitrary_unit,
/Users/bsipocz/.pyenv/versions/3.5.8/lib/python3.5/site-packages/astropy/units/quantity_helper.py:457: in <module>
    sps.erf, sps.gamma, sps.loggamma, sps.gammasgn,
E   AttributeError: module 'scipy.special' has no attribute 'loggamma'
```

**Comment by bsipocz** [comment]
(For master we need to require an even newer version, with a very quick try I'm having trouble installing 0.18 for our supported versions of python (and wheels are not available for that for >python 3.5)

**Comment by bsipocz** [comment]
(On the other hand, requiring a new scipy version just to provide `loggamma` suport for units seems a bit of an overreaction, but this ship has been sailed, we must simply need to be better for declaring and testing dependencies)
